fdbclient/admin_client.go

```diff
@@ -823,3 +823,40 @@ func (client *cliAdminClient) SetProcessesUnderMaintenance(processGroupIDs []fdb
 
 	return err
 }
+
+// GetVersionFromReachableCoordinators will return the running version based on the reachable coordinators. This method
+// can be used during version incompatible upgrades and based on the responses of the coordinators, this method will
+// assume the current running version of the cluster. If the fdbcli calls for none of the provided version return
+// a majority of reachable coordinators, an empty string will be returned.
+func (client *cliAdminClient) GetVersionFromReachableCoordinators() string {
+	// First we test to get the status from the fdbcli with the current running version defined in cluster.Status.RunningVersion.
+	status, _ := client.getStatusFromCli(false)
+	if quorumOfCoordinatorsAreReachable(status) {
+		return client.Cluster.GetRunningVersion()
+	}
+
+	// If the majority of coordinators are not reachable with the cluster.Status.RunningVersion, we try the desired version
+	// if the cluster is currently performing an version incompatible upgrade.
+	if client.Cluster.IsBeingUpgradedWithVersionIncompatibleVersion() {
```

Comment:
Is this changing the version the client uses (or where it tries to read the status) so that if it works to get status then we know it's updated even if it doesn't have running processes? I don't really follow

Reply:
Discussed offline, it makes sense now but it would be good to add a comment here that setting the running version as the desired one `checkClientCompatibility` gets skipped and allows the status to be got

Reply:
This would be the wrong place for the comment as this part will only change the running version for the client's cluster copy to force it to use the newer fdbcli version. The actual change is done in the cluster controller where we already have a comment:

https://github.com/FoundationDB/fdb-kubernetes-operator/pull/2098/files#diff-1ca08b6f9e9610ef14189184290a644a29adbb47e6f8653a479d75fdce4a8452R602-R604